deepsensor/model/model.py

```diff
@@ -811,147 +835,265 @@ def get_index(*args, x1 = True) -> Union[int, Tuple[List[int], List[int]]]:
+                            patch_prev = preds[i - patches_per_row]
+                            prev_patch_x1_max = get_index(
+                                int(
+                                    patch_prev[var_name]
+                                    .coords[unnorm_coord_names["x1"]]
+                                    .max()
+                                ),
+                                x1=True,
+                            )
+                            b_x1_min = (
+                                prev_patch_x1_max - patch_x1_index[0]
+                            ) - patch_overlap[0]
 
                         patch_clip_x1_min = int(b_x1_min)
-                        patch_clip_x1_max = int(data_array.sizes['y'] - b_x1_max)
+                        patch_clip_x1_max = int(
+                            data_array.sizes[unnorm_coord_names["x1"]] - b_x1_max
+                        )
                         patch_clip_x2_min = int(b_x2_min)
-                        patch_clip_x2_max = int(data_array.sizes['x'] - b_x2_max)
+                        patch_clip_x2_max = int(
+                            data_array.sizes[unnorm_coord_names["x2"]] - b_x2_max
+                        )
 
-                        patch_clip = data_array.isel(y=slice(patch_clip_x1_min, patch_clip_x1_max),
-                                                    x=slice(patch_clip_x2_min, patch_clip_x2_max))
+                        patch_clip = data_array[
+                            {
+                                unnorm_coord_names["x1"]: slice(
+                                    patch_clip_x1_min, patch_clip_x1_max
+                                ),
+                                unnorm_coord_names["x2"]: slice(
+                                    patch_clip_x2_min, patch_clip_x2_max
+                                ),
+                            }
+                        ]
 
                         patches_clipped[var_name].append(patch_clip)
 
-            combined = {var_name: xr.combine_by_coords(patches, compat='no_conflicts') for var_name, patches in patches_clipped.items()}
+            combined = {
+                var_name: xr.combine_by_coords(patches, compat="no_conflicts")
+                for var_name, patches in patches_clipped.items()
+            }
 
             return combined
 
+        # sanitise patch_size and stride arguments
+
+        if isinstance(patch_size, float) and patch_size is not None:
+            patch_size = (patch_size, patch_size)
+
+        if isinstance(stride, float) and stride is not None:
+            stride = (stride, stride)
+
+        if stride[0] > patch_size[0] or stride[1] > patch_size[1]:
+            raise ValueError(
+                f"stride must be smaller than patch_size in the corresponding dimensions. Got: patch_size: {patch_size}, stride: {stride}"
+            )
+
+        for val in list(stride + patch_size):
```

Comment:
I think recall from Nils' previous PR that the normalised x1/x2 coordinates can take values >1.0. So may need to consider this. However, this may also be fine, because I don't think values can be >>1.0, meaning that I think it would be rare for people to genuinely want to use a patch or stride side >1.0.